### ISSUE TITLE
C901 complex-structure has unclear documentation

### ISSUE THREAD

**Issue by GrantMoyer** [astral-sh/ruff#25028]
### Summary

The bad

```python
def foo(a, b, c):
    if a:
        if b:
            if c:
                return 1
            else:
                return 2
        else:
            return 3
    else:
        return 4
```

and good

```python
def foo(a, b, c):
    if not a:
        return 4
    if not b:
        return 3
    if not c:
        return 2
    return 1
```

examples for C901 have the same McCabe complexity (4), so the lint would also trigger for the good example if it would trigger for the bad example.
